tests/refs/reflog/reflog_helpers.c

```diff
@@ -15,51 +15,107 @@ size_t reflog_entrycount(git_repository *repo, const char *name)
 	return ret;
 }
 
-void reflog_check_entry(git_repository *repo, const char *reflog, size_t idx,
+void cl_reflog_check_entry_(git_repository *repo, const char *reflog, size_t idx,
 	const char *old_spec, const char *new_spec,
-	const char *email, const char *message)
+	const char *email, const char *message, const char *file, int line)
 {
 	git_reflog *log;
 	const git_reflog_entry *entry;
+	git_buf result = GIT_BUF_INIT;
 
 	cl_git_pass(git_reflog_read(&log, repo, reflog));
 	entry = git_reflog_entry_byindex(log, idx);
+	if (entry == NULL) {
+		clar__fail(file, line, "Reflog has no such entry", NULL, 1);
+	}
 
 	if (old_spec) {
 		git_object *obj = NULL;
 		if (git_revparse_single(&obj, repo, old_spec) == GIT_OK) {
-			cl_assert_equal_oid(git_object_id(obj), git_reflog_entry_id_old(entry));
+			if (git_oid_cmp(git_object_id(obj), git_reflog_entry_id_old(entry)) != 0) {
+				char err[] = "\"........................................\" != \"........................................\"";
+
+				git_oid_fmt(&err[1], git_object_id(obj));
+				git_oid_fmt(&err[47], git_reflog_entry_id_old(entry));
+
+				git_buf_printf(&result, "\tOld OID: %s\n", err);
+			}
 			git_object_free(obj);
 		} else {
 			git_oid *oid = git__calloc(1, sizeof(*oid));
 			git_oid_fromstr(oid, old_spec);
-			cl_assert_equal_oid(oid, git_reflog_entry_id_old(entry));
+			if (git_oid_cmp(oid, git_reflog_entry_id_old(entry)) != 0) {
+				char err[] = "\"........................................\" != \"........................................\"";
+
+				git_oid_fmt(&err[1], oid);
+				git_oid_fmt(&err[47], git_reflog_entry_id_old(entry));
+
+				git_buf_printf(&result, "\tOld OID: %s\n", err);
+			}
 			git__free(oid);
 		}
 	}
 	if (new_spec) {
 		git_object *obj = NULL;
 		if (git_revparse_single(&obj, repo, new_spec) == GIT_OK) {
-			cl_assert_equal_oid(git_object_id(obj), git_reflog_entry_id_new(entry));
+			if (git_oid_cmp(git_object_id(obj), git_reflog_entry_id_new(entry)) != 0) {
+				char err[] = "\"........................................\" != \"........................................\"";
+
+				git_oid_fmt(&err[1], git_object_id(obj));
+				git_oid_fmt(&err[47], git_reflog_entry_id_new(entry));
+
+				git_buf_printf(&result, "\tNew OID: %s\n", err);
```

Comment:
Huh, this seems a bit fragile. How about

```
git_oid__writebuf(&result, "\tNew OID: \"", git_object_id(obj));
git_oid__writebuf(&result, "\" != \", git_reflog_entry_id_new(entry));
git_buf_puts(&result, '"\n');
```

This would save us from the hardcoded indices. Same applies below.